bgpd/bgp_attr.c

```diff
@@ -1603,7 +1606,10 @@ bgp_attr_community(struct bgp_attr_parser_args *args)
 	/* XXX: fix community_parse to use stream API and remove this */
 	stream_forward_getp(peer->curr, length);
 
-	if (!attr->community)
+	/* The Community attribute SHALL be considered malformed if its
+	 * length is not a non-zero multiple of 4.
+	 */
+	if (!attr->community || length % 4)
```

Comment:
community_parse already has check for 

	if (length % 4)
		return NULL;
adding extra check length % 4 might not be required IMHO as that or condition will never be true.

Reply:
you are right